Release fails if there are no changes to gh-pages branch

Use case: My azure setup builds on every merge to master and releases on every master build. If I merge in a change to a readme file or `.gitignore` or something else that results in no changes to HTML, the release will fail. It would be nice if this situation doesn't cause the task to fail. The logs:
```
2018-11-15T05:25:47.6589352Z ##[section]Starting: Publishes to GitHub Pages
2018-11-15T05:25:47.6594060Z ==============================================================================
2018-11-15T05:25:47.6594159Z Task         : Publish to GitHub Pages
2018-11-15T05:25:47.6594218Z Description  : Publishes files to a GitHub Pages gh-pages branch
2018-11-15T05:25:47.6594270Z Version      : 0.13.0
2018-11-15T05:25:47.6594334Z Author       : James Randall
2018-11-15T05:25:47.6594382Z Help         : 
2018-11-15T05:25:47.6594451Z ==============================================================================
2018-11-15T05:25:48.7385928Z Cloning existing GitHub Pages branch
2018-11-15T05:25:53.1698006Z Copying new documentation into branch
2018-11-15T05:25:53.2787015Z Committing the GitHub Pages Branch
2018-11-15T05:25:53.4661368Z On branch gh-pages
2018-11-15T05:25:53.4662152Z Your branch is up to date with 'origin/gh-pages'.
2018-11-15T05:25:53.4662393Z 
2018-11-15T05:25:53.4662676Z nothing to commit, working tree clean
2018-11-15T05:25:53.4791413Z ##[error]Error committing - see earlier log, error code 1
2018-11-15T05:25:55.7297391Z ##[error]Exit code 1 returned from process: file name 'C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe', arguments '-NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". ([scriptblock]::Create('if (!$PSHOME) { $null = Get-Item -LiteralPath ''variable:PSHOME'' } else { Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Management\Microsoft.PowerShell.Management.psd1'')) ; Import-Module -Name ([System.IO.Path]::Combine($PSHOME, ''Modules\Microsoft.PowerShell.Utility\Microsoft.PowerShell.Utility.psd1'')) }')) 2>&1 | ForEach-Object { Write-Verbose $_.Exception.Message -Verbose } ; Import-Module -Name 'D:\_work\_tasks\GitHubPagesPublish_65386086-c58c-4396-bc83-4c7a39e84448\0.13.0\ps_modules\VstsTaskSdk\VstsTaskSdk.psd1' -ArgumentList @{ NonInteractive = $true } -ErrorAction Stop ; $VerbosePreference = 'SilentlyContinue' ; $DebugPreference = 'SilentlyContinue' ; Invoke-VstsTaskScript -ScriptBlock ([scriptblock]::Create('. ''D:\_work\_tasks\GitHubPagesPublish_65386086-c58c-4396-bc83-4c7a39e84448\0.13.0\publish.ps1'''))"'.
2018-11-15T05:25:55.7302520Z ##[section]Finishing: Publishes to GitHub Pages
```

--- a/buildAndReleaseTask/publish.ps1
+++ b/buildAndReleaseTask/publish.ps1
@@ -36,7 +36,7 @@ try {
     git config user.email $githubemail
     git config user.name $githubusername
     git add *
-    git commit -m $commitMessage
+    git commit --allow-empty -m $commitMessage
 
     if ($lastexitcode -gt 0)
     {

